addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/handler/BridgeHandler.java

```diff
@@ -0,0 +1,382 @@
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+import org.joda.time.LocalTime;
+import org.openhab.binding.tankerkoenig.TankerkoenigBindingConstants;
+import org.openhab.binding.tankerkoenig.internal.config.LittleStation;
+import org.openhab.binding.tankerkoenig.internal.config.OpeningTime;
+import org.openhab.binding.tankerkoenig.internal.config.OpeningTimes;
+import org.openhab.binding.tankerkoenig.internal.config.TankerkoenigListResult;
+import org.openhab.binding.tankerkoenig.internal.data.TankerkoenigService;
+import org.openhab.binding.tankerkoenig.internal.utility.Holiday;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BridgeHandler} is responsible for handling the things (Tankstellen)
+ *
+ *
+ * @author Dennis Dollinger/Jürgen Baginski
+ */
+public class BridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private String apiKey;
+    private int refreshInterval;
+    private boolean setupMode;
+    private boolean use_OpeningTime;
+
+    private ArrayList<Thing> tankstellenThingList;
+    private HashMap<String, LittleStation> tankstellenList;
+
+    private TankerkoenigListResult tankerkoenigListResult;
+
+    private ScheduledFuture<?> pollingJob;
+
+    public BridgeHandler(Bridge bridge) {
+        super(bridge);
+        tankstellenThingList = new ArrayList<Thing>();
+        tankstellenList = new HashMap<String, LittleStation>();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no code needed.
+    }
+
+    @Override
+    public void initialize() {
+        Configuration config = getThing().getConfiguration();
+        setApiKey((String) config.get(TankerkoenigBindingConstants.CONFIG_API_KEY));
+        setRefreshInterval(((BigDecimal) config.get(TankerkoenigBindingConstants.CONFIG_REFRESH)).intValue());
+        setSetupMode((boolean) config.get(TankerkoenigBindingConstants.CONFIG_SETUP_MODE));
+        setUseOpeningTime((boolean) config.get(TankerkoenigBindingConstants.CONFIG_USE_OPENINGTIME));
+
+        // updateStatus(ThingStatus.ONLINE);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        int pollingPeriod = this.getRefreshInterval();
+        pollingJob = scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Try to refresh data");
+                // Just update data if setupMode is false(off)
+                try {
+                    if (!isSetupMode()) {
+                        updateTankstellenData();
+                        updateTankstellenThings();
+                    }
+                } catch (Throwable t) {
```

Comment:
What I'm trying to do: The scheduler will get cancled for any exception that isn't caugth. Handling all eceptions in any way will keep the scheduler running.
I had a lost internet-connection while the scheduler tried to get data, that canceld the whole scheduler, having a try catch, the scheduler would continue and try next time again.
I'll change it to RuntimeException, if you have a better suggestion, I'll take it.